keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1428 @@
+          localObjectReference: kube-root-ca.crt
+          items:
+          - key: ca.crt
+            path: kube-apiserver-root-certificate.pem
+      - downwardAPI:
+          items:
+          - path: namespace
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+```
+
+I ensure that my pod uses client-go's InClusterConfig, and client-go automatically detects the presence uses the mTLS client certificate to authenticate to kube-apiserver.
+
+#### Using this feature as a third-party signer implementer
+
+### Future expansion: HSM support for private keys
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+Unit test list (informed by draft implementation):
+* pkg/api/pod/util_test.go
+* pkg/apis/certificates/validation_test.go
+* pkg/apis/core/validation/validation_test.go
+* pkg/controller/certificates/authority/authority_test.go
+* pkg/controller/certificates/podcertificaterequest_controller_test.go
+* pkg/controller/certificates/signer/signer_test.go
+* pkg/kubelet/podcertificate/podcertificate_manager_test.go
+* pkg/registry/certificates/podcertificaterequest/storage/storage_test.go
+* pkg/registry/certificates/podcertificaterequest/strategy_test.go
+* pkg/volume/projected/projected_test.go
+* plugin/pkg/admission/noderestriction/admission_test.go
+* plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy_test.go
+* plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy_test.go
+* staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509_test.go
+* staging/src/k8s.io/client-go/rest/config_test.go
+* staging/src/k8s.io/component-helpers/kubernetesx509/kubernetesx509_test.go
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
```

Comment:
Nit: but the above comment explicitly requires both the directory and coverage :sweat_smile:

Reply:
The coverage tab crashes in Chrome :/ I need to dig up a Firefox install

Reply:
You can always just run `go test -race -cover <package>` :wink:

Reply:
Will do.